src/components/Menu/DrawerMenu.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2023 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { ComponentPropsWithoutRef, ReactNode, forwardRef } from "react";
+import styles from "./DrawerMenu.module.css";
+import { platform } from "../../utils/platform";
+import classNames from "classnames";
+
+interface Props extends ComponentPropsWithoutRef<"div"> {
+  /**
+   * The menu title.
+   */
+  title: string;
```

Comment:
I do think we should try to always give menus a title, because they require an accessible label under ARIA. We could give them a generic label like "Menu" in the case where they have no title, but I think that's against the spirit of the spec.

However, maybe the takeaway from Element Web is that menus don't always need to have a *visible* label, in which case I could split this up into a `label` and `title` prop.